### ISSUE TITLE
Numeric Filters not working with Druid

### ISSUE THREAD

**Issue by denisesenguel** [metabase/metabase#10935]
**Description**

None of the query builder filters for numeric columns (greater than, equal to, is empty, …) works for Druid datasources (for Postgres tables they are working just fine)

The filters are simply ignored and all rows returned without any error being thrown. 
When switching from the query builder to a native query, it becomes apparent that the filter does not appear in the query at all (see Example for details)

This happens regardless of whether the column is defined as a metric or a dimension of type long in Druid upon ingestion.

**Example**

When selecting any filter for a numeric column and grouping by the same column the problem is clearly visible (the same problem occurs for raw data or any type of aggregation, I just found this example best for illustrating the issue)
![Bildschirmfoto 2019-09-19 um 13 12 07](https://user-images.githubusercontent.com/17010657/65240007-8c20cc80-dae0-11e9-98b8-e0d0aeedf520.png)
Sorry the screenshot is in german "größer als" means "greater than"!

Looking at the native query of this built question reveals that there actually is no filter 😱 
![Bildschirmfoto 2019-09-19 um 13 13 05](https://user-images.githubusercontent.com/17010657/65240026-9773f800-dae0-11e9-8826-2082954c1d0b.png)

When adding the following filter to the query
```
"filter": {
        "type": "bound",
        "dimension": "screen_width",
        "lower": "400" ,
        "ordering": "numeric"
    }
```
The correct result is returned! ✨ 

**Installation Details**

I’m using Chrome on Mac OS Mojave 10.14.1

Datasource is Druid 0.15.0

We have Metabase 0.33.2 hosted using Docker and with an internal Postgres Database.

**Severity**

Filtering rows e.g. by revenue is a very common Usecase and Druid our main Datasource, so this is quite limiting for us at the moment.
It also seems to me that the issue isn’t terribly hard to fix! 🙃

**Comment by viveksatasiya** [comment]
I found the same issue while filtering value based on column with data type long in druid.

**Comment by sbelak** [comment]
Seems like 2 things are happening:
1) we always treat numeric columns as metrics (which is wrong); and
2) we currently don't support filtering on metrics 

1) is also a (partial) culprit for #10617.

**Comment by kkalyan** [comment]
In the latest version, the update filter button is disabled for Druid queries.

![image](https://user-images.githubusercontent.com/391567/75940160-1a93cb00-5e41-11ea-90e3-cb4cdf7735f8.png)

I think right way to do this in Druid is by applying `having` filter on groupby

```
"having":
        {
            "type": "greaterThan",
            "aggregation": "<aggregate_metric>",
            "value": <numeric_value>
        }
```
https://druid.apache.org/docs/latest/querying/having.html

@salsakran @camsaul I'm happy to submit a PR, if the approach sounds good to you.

**Comment by denisesenguel** [comment]
I cannot reproduce this with 0.34.3 @kkalyan, for me nothing has changed compared to previous versions. Filters applied to numeric dimensions are still simply ignored. 

Your code snipped applies a filter based on an aggregated metric in a groupBy query, I don't see how this can be implemented in the query builder in the near future. I think it goes one step further. I would be happy with just getting standard row filtering based on dimensions of type LONG or FLOAT as outlined in the initial issue report.

**Comment by denisesenguel** [comment]
Nice, this seems to be fixed with version 0.35.0, filters are working for the first time! 👏 💯 

However, graphs show the complete range of a numeric dimension although data has been filtered by it (this does NOT apply to tables)

![Bildschirmfoto 2020-04-06 um 12 25 00](https://user-images.githubusercontent.com/17010657/78548862-baa69080-7801-11ea-95db-41add28afd55.png)

In the depicted case a "less than 30" filter was applied on the dimension on the x-axis.
This is quite bad as there is no way to restrict axes manually.

**Comment by camsaul** [comment]
This was fixed by #11992

Additional tests added by #10935
